### PR TITLE
enh(sec) Improve authentication messages in login.log

### PR DESCRIPTION
**Pull request by CPbN**
Hi,

## Description

This PR follows PR #7502 (not merged).
Goal is for the system administrator to be able to use tools such as fail2ban to ban IPs generating too many failed login attempts.
First step is then to have correctly formatted authentication messages.
This PR is then rather simple.

Related Fail2Ban PR is here (already merged) : https://github.com/fail2ban/fail2ban/pull/2550

## Type of change

- [x] Patch fixing an issue (non-breaking change)

## Target serie

- [x] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Generate failed logins and have a look into `/var/log/centreon/login.log`.

Thank you :+1:

Edit : still missing from 19.10.4.

### CODE REVIEW

**Comment by CPbN** [comment]
Corresponding Fail2Ban PR merged, good news 👍

File: www/class/centreonLog.class.php [hunk 1]
```diff
@@ -82,8 +82,9 @@ public function insertLog($id, $str, $print = 0, $page = 0, $option = 0)
     {
         /*
          * Construct alerte message
```
Comment (sc979) [inline www/class/centreonLog.class.php]:
```suggestion
         * Construct alert message
```

**Comment by CPbN** [comment]
Thank you again @sc979 for this review :+1:
I'll push a commit to answer your suggestions above ASAP :wink:

File: www/class/centreonAuth.class.php [hunk 2]
```diff
@@ -319,19 +319,21 @@ protected function checkUser($username, $password, $token)
...
                         1,
-                        "[".$this->source."] Contact '" . $username . "' logged in - IP : " . $_SERVER["REMOTE_ADDR"]
+                        "[" . $this->source . "] [" . $_SERVER["REMOTE_ADDR"] . "] Authentication succeeded for '" . $username . "'"
                     );
                 } else {
+                    //  Take care before modifying this message pattern as it may break tools such as fail2ban
                     $this->CentreonLog->insertLog(
                         1,
-                        "Contact '" . $username . "' doesn't match with password"
+                        "[" . $this->source . "] [" . $_SERVER["REMOTE_ADDR"] . "] Authentication failed for '" . $username . "' : password mismatch"
```
[the remark this answers is not in the document]
Comment (CPbN) [reply www/class/centreonAuth.class.php]: Done.

[the remark this answers is not in the document]
Comment (CPbN) [reply www/class/centreonAuth.class.php]: Done.

File: www/class/centreonAuth.class.php [hunk 3]
```diff
@@ -319,19 +319,21 @@ protected function checkUser($username, $password, $token)
...
                     );
                     $this->error = _('Your credentials are incorrect.');
                 }
             } else {
+                //  Take care before modifying this message pattern as it may break tools such as fail2ban
                 $this->CentreonLog->insertLog(
                     1,
-                    "[".$this->source."] Contact '" . $username . "' is not enable for reaching centreon"
+                    "[" . $this->source . "] [" . $_SERVER["REMOTE_ADDR"] . "] Authentication failed for '" . $username . "' : not authorized"
```
[the remark this answers is not in the document]
Comment (CPbN) [reply www/class/centreonAuth.class.php]: > In this case, the user exist and his auth data are correct

That was not the case.
I then now check for correct password before checking whether or not user is allowed to access Centreon. Which is better IMO in terms of security.
Also then back to previous message pattern here.

File: www/class/centreonLog.class.php (hunk 1 above)
Comment (CPbN) [reply www/class/centreonLog.class.php]: Done.

File: www/class/centreonAuth.class.php [hunk 4]
```diff
@@ -366,10 +368,11 @@ protected function checkUser($username, $password, $token)
                 }
             }
         } else {
-            if ($this->debug) {
+            if (($this->debug) || (strlen($username) > 0)) {
```
[the remark this answers is not in the document]
Comment (CPbN) [reply www/class/centreonAuth.class.php]: This is what this PR is intended for, having these failed login attempts messages in log, whatever login used, existing or not.
Goal is to ban IPs doing brute force attempts / login attempts, for example when Centreon is available online. We then want these messages to feed Fail2Ban :wink:

[the remark this answers is not in the document]
Comment (sc979) [reply www/class/centreonAuth.class.php]: Iindeed. Then we need to remove the debug flag, as it becomes useless

Comment (sc979) [inline www/class/centreonAuth.class.php]:
```suggestion
            if (strlen($username) > 0) {
```

[the remark this answers is not in the document]
Comment (CPbN) [reply www/class/centreonAuth.class.php]: Done, thx :+1:

Comment (CPbN) [reply www/class/centreonAuth.class.php]: Done.

**Comment by CPbN** [comment]
Let's merge this tiny PR when you'll have a few minutes ?
Thank you very much :)

**Comment by sc979** [comment]
@cgagnaire , @kduret, @lpinsivy   -> I've tested it and everything seems fine. I'm merging it

**Comment by sc979** [comment]
@CPbN , thanks for your work

**Comment by CPbN** [comment]
Good news, many thanks too @sc979 👍

**Pull request merged**